### ISSUE TITLE
Getting Segmentation fault when profiling cpp program in rasbperry pi 5 

### ISSUE THREAD

**Issue by mathanprasannakumar** [wolfpld/tracy#873]
Hi @wolfpld , 

Thank you for providing this awesome profiler , i am having a simple cpp program which runs a simple while loop  when i run this program on the x86  to profile , i was able to profile this cpp program but when i run the same program on raspberry pi i am getting segmentation fault . 

Below is the code 

```
#include <iostream>
#include "tracy/Tracy.hpp"

int main()
{
    while(true)
    {
    ZoneScoped; 
    }
    return 0;
}
```

raspberry pi 5 details: 
`Linux raspberrypi 6.6.31+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.31-1+rpt1 (2024-05-29) aarch64 GNU/Linux`

Please let me know what is the issue and what can i do to profile a cpp program in armv8 architecture(raspberry pi 5) ?

**Comment by TheHoodedFoot** [comment]
I have encountered the same issue when attempting to profile my code on a Raspberry Pi 5.

A quick examination shows that public/client/TracyProfiler.cpp reads /sys/devices/system/cpu/cpu#/topology/physical_package_id on Linux systems, and if it finds the endpoint then it assumes that the core_id and die_id endpoints also exist, and doesn't check when opening them. However, on the Raspberry Pi 5 (and 4 it seems) the die_id endpoint is missing, causing the segmentation fault.

As a gruesome hack I have managed to get my code compiling by changing TracyProfiler.cpp to just use the core_id endpoint twice, but I don't know enough to go any further.

I am using the current master branch (ea4de3be).
